test/hipcub/test_hipcub_block_load_store.cpp

```diff
@@ -481,3 +489,196 @@ TYPED_TEST(HipcubBlockLoadStoreClassTests, LoadStoreClassDefault)
         HIP_CHECK(hipFree(device_output));
     }
 }
+
+
+template <bool IF, typename ThenType, typename ElseType>
+struct If
+{
+    /// Conditional type result
+    typedef ThenType Type;      // true
+};
```

Comment:
This conditional implementation is bugged, as it can never return the `ElseType`. Please use [`std::conditional`](https://en.cppreference.com/w/cpp/types/conditional) instead.

Reply:
And I had to replace the rocprim implementation alsoe because now the test working correctly and the rocprim iterator value type is not void.